HHS/Head-Start-TTADP
# [Prod] Deploy similiarity API, performance fixes for queue, fix bug with session storage & RTR, lock down completed sessions

## Description of change

### TTAHUB-2164

**When the session is complete**
- Hide the "edit" and "delete" buttons from the interface
- Prevent access to the session form

### TTAHUB-2086

- Fixes a bug where a filtering a recipient's data by a grant number that either doesn't exist for them or hasn't been loaded yet returns a white screen
- Also changed the key used by session storage to save the filters for that page - since it wasn't recipient specific, a grant for one recipient could be loaded for another, which I believe is how the bug happens.

This PR also deploys the goal similarity API to production (no way to test) and includes fixes to the queuing system to make errors less aggressively consume memory

## How to test

- **TTAHUB-2164**: Confirm that a completed session no longer has "edit" or "delete" buttons

- **TTAHUB-2086**: Visit a URL like this, with nonsense in the grantNumber.in filter:
/recipient-tta-records/564/region/3/goals-objectives?grantNumber.in%5B%5D=ASDF

It should not white screen. Select a grant for that recipient that does exist, and, as long as that works, visit another recipient's RTTAPA page that you haven't visited in the session yet. Confirm that no out-of-scope grant number filters are loaded


## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2164
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2086
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2146
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1990

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
